react/components/Statement/index.tsx

```diff
@@ -3,36 +3,39 @@ import React from 'react'
 import SubjectAtom, { SubjectOptions } from './Atoms/SubjectAtom'
 import VerbAtom from './Atoms/VerbAtom'
 import ObjectAtom from './Atoms/ObjectAtom'
+import { StatementProp } from './typings'
 
-type Props = {
+type StatementProps = {
   isFullWidth?: boolean
   isRtl?: boolean
   omitSubject?: boolean
   omitVerbs?: boolean
-  onChangeStatement: (statement: Props['statement']) => void
+  onChangeStatement: (statement: StatementProp) => void
   options: SubjectOptions
-  statement?: {
-    subject: string
-    verb: string
-    object?: unknown
-    error?: string
-  }
+  statement?: StatementProp
   subjectPlaceholder: string
 }
 
-const Statement: React.FC<Props> = ({
+const NoopComponent = () => <></>
+
+const Statement: React.FC<StatementProps> = ({
   isFullWidth = false,
   isRtl,
   omitSubject,
   omitVerbs,
   onChangeStatement,
   options,
-  statement = { subject: '', verb: '', object: null, error: null },
+  statement = { subject: '', verb: '', object: null },
   subjectPlaceholder,
 }) => {
-  const verbOptions =
-    statement.subject &&
-    options[statement.subject].verbs.find(verb => verb.value === statement.verb)
+  const verbOptions = (Boolean(statement.subject) &&
+    options[statement.subject].verbs.find(
+      verb => verb.value === statement.verb
+    )) || {
+    label: '',
+    value: '',
+    object: NoopComponent,
+  }
```

Comment:
Boy what is happening here 😆 

I'd rewrite this to something a bit more readable, like:

```suggestion
let verbOptions = {
	label: '',
	value: '',
	object: NoopComponent,
};

if (statement.subject) {
	const foundOption = options[statement.subject].verbs.find(
		(verb) => verb.value === statement.verb,
	);

	if (foundOption) {
		verbOptions = foundOption;
	}
}
```

Reply:
Fixed